chipy_org/apps/meetings/models.py

```diff
@@ -276,17 +331,85 @@ def clean(self):
     def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
         self.full_clean()
 
+        original = None
+        if self.id:
+            original = RSVP.objects.get(id=self.id)
+
         # Generate a key for this RSVP
         if not self.key:
             self.key = "".join(
                 random.choice(string.digits + string.ascii_lowercase) for x in range(40)
             )
 
-        return super(RSVP, self).save(*args, **kwargs)
+        if not (original and original.response == self.response):
+            if self.response == self.Responses.IN_PERSON:
+                if self.meeting.has_in_person_capacity():
+                    self.status = self.Statuses.CONFIRMED
+                else:
+                    self.status = self.Statuses.WAIT_LISTED
+
+            if self.response == self.Responses.VIRTUAL:
+                if self.meeting.has_virtual_capacity():
+                    self.status = self.Statuses.CONFIRMED
+                else:
+                    self.status = self.Statuses.WAIT_LISTED
+
+        if self.response == self.Responses.DECLILNED:
+            self.status = self.Statuses.CONFIRMED
+
+        return super().save(*args, **kwargs)
 
     @property
     def full_name(self):
         return self.first_name + " " + self.last_name
 
     def __str__(self):
         return f"{self.meeting}: {self.full_name}"
+
+
+@receiver(post_save, sender=RSVP)
+def rsvp_post_save(sender, instance, **kwargs):
+
+    # don't run on bulk or raw updates
+    if kwargs["raw"]:
+        return
+
+    meeting: Meeting = instance.meeting
+
+    # don't updated anything if registration is closed
+    # if not meeting.can_register():
+    #     return
+
+    # send an email to the user by email
+    if instance.email:
+        send_rsvp_email(instance)
+
+    if meeting.has_in_person_capacity():
+        first_on_in_person_wait_list = (
```

Comment:
I would follow the "fat models" approach and move this into the meeting object. `meeting.allow_waitlisted_individuals(1)` or something like that seems reasonable.

It would help decouple your hook code from the logic to manage the RSVPs.

Reply:
That said, this is clearly functional :D